src/Nethermind/Nethermind.Core/Specs/SpecProviderDecorator.cs

```diff
@@ -0,0 +1,31 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Int256;
+
+namespace Nethermind.Core.Specs;
+
+public class SpecProviderDecorator(ISpecProvider baseSpecProvider) : ISpecProvider
```

Comment:
Is there a point for this class existence? We have only one usage and would have to make other things virtual if needed. Seems not needed to me.